Foundation/NSStream.swift

```diff
@@ -209,20 +227,32 @@ public class NSOutputStream : Stream {
         return self.init(toMemory: ())
     }
     
-    public override func propertyForKey(_ key: String) -> AnyObject? {
-        return CFWriteStreamCopyProperty(_stream, key._cfObject)
+    public override var streamError: NSError?{
+        let error = CFWriteStreamCopyError(_stream)
+        return error?._nsObject
     }
     
     public  override func setProperty(_ property: AnyObject?, forKey key: String) -> Bool {
+        guard let property = property else { return false }
         return CFWriteStreamSetProperty(_stream, key._cfObject, property)
     }
+    
+    public override func propertyForKey(_ key: String) -> AnyObject? {
+        return CFWriteStreamCopyProperty(_stream, key._cfObject)
+    }
+    
 }
 
 // Discussion of this API is ongoing for its usage of AutoreleasingUnsafeMutablePointer
 #if false
 extension Stream {
+    //Currently not tested as it is compiled out through the macro setting false
```

Comment:
My apologies if I am misinterpreting, but the CFNetwork Programing Guide lead me to think that situation could exist.

https://developer.apple.com/library/ios/documentation/Networking/Conceptual/CFNetwork/CFStreamTasks/CFStreamTasks.html#//apple_ref/doc/uid/TP30000230-62233

The CFStreamCreatePairWithSocketToCFHost function returns the new read stream reference in myReadStream. The last parameter, NULL, indicates that the caller does not want to create a write stream. If you wanted to create a write steam, the last parameter would be, for example, &myWriteStream.
https://developer.apple.com/library/mac/documentation/CoreFoundation/Reference/CFStreamConstants/#//apple_ref/c/func/CFStreamCreatePairWithSocketToHost

readStream: If you pass NULL, this function will not create a readable stream
writeStream: If you pass NULL, this function will not create a writable stream